apis/v1beta1/shared_types.go

```diff
@@ -43,6 +44,8 @@ type ParentReference struct {
 	//
 	// Support: Core (Gateway)
 	//
+	// Support: Core (Service - mesh conformance profile only)
+	//
 	// Support: Implementation-specific (Other Resources)
```

Comment:
Maybe something similar to what I suggested above.

```suggestion
	// There are two kinds of parent resources with "Core" support:
	//
	// * Gateway (Gateway conformance profile)
	// * Service (Mesh conformance profile)
	//
	// Support for other resources is Implementation-Specific.
```